### PR TITLE
Add Cloud Input Function

### PR DESCRIPTION
**Pull request by Inokinoki**
WIP

### CODE REVIEW

**Comment by epico** [comment]
Could you update the commits with the following requirements?

1. Please use `git rebase <branch>` to upstream master branch
   a. The command will make your commit log shorter
2. Please squash the git commits
   a. maybe one commit for each file change or two file changes
   b. better to include header file and C++ file in one single commit
3. Please don't include po changes for code review

This will make the code review easier, thanks in advance!
I think `git rebase -i` will help to update the code.

Feel free to create another pull request, if you prefer. :)

**Comment by Inokinoki** [comment]
> > WIP
> 
> Could you please clean up these commits? they are hard to be reviewed.

Sure. Sorry for the hardness in review... I just wanted to keep the previous commits in this branch.

File: src/PYPCloudCandidates.cc [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+        enhanced.m_display_string = CANDIDATE_PENDING_TEXT;
+        enhanced.m_candidate_type = CANDIDATE_CLOUD_INPUT;
+        m_candidates.push_back (enhanced);
+    }
+    candidates.insert (candidates.begin () + m_first_cloud_candidate_position - 1, m_candidates.begin (), m_candidates.end ());
+
+    if (! m_editor->m_config.doublePinyin ())
+    {
+        const gchar *text = m_editor->m_text;
+        if (strlen (text) >= m_min_cloud_trigger_length)
```
[the remark this answers is not in the document]
Comment (epico) [reply src/PYPCloudCandidates.cc]: Sorry, I prefer to put "#include <cstring>" in PYPCloudCandidates.cc.

[the remark this answers is not in the document]
Comment (epico) [reply src/PYPCloudCandidates.cc]: Sorry, I prefer to put "#include \<cstring\>" in PYPCloudCandidates.cc.

**Comment by qiangzhao** [comment]
I will agree with the above.

**Comment by Inokinoki** [comment]
Replaced by #229

**Pull request closed without merging**